### PR TITLE
fix(toolbar): add script-src directive to CSP header

### PR DESCRIPTION
**Pull request by aliu39**
This allows the browser to run the inline scripts in the toolbar templates (login-success.html, iframe.html, etc). We set this header in prod already - not sure where. Hopefully prod doesn't override it.. still need more info on this.

### CODE REVIEW

File: src/sentry/toolbar/views/base.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+class ToolbarView(OrganizationView):
+    def respond(self, template: str, context: dict[str, Any] | None = None, status: int = 200):
+        response = super().respond(template, context=context, status=status)
+        # Allow running the inline scripts in the response templates
+        response["Content-Security-Policy"] = csp_add_directive(
+            response.get("Content-Security-Policy", ""),
+            "script-src",
+            ["'unsafe-inline'", "sentry.io", "*.sentry.io"],
```
Comment (ryan953) [inline src/sentry/toolbar/views/base.py:18]: hard-coded domains, this will need to be tweaked to support on-prem installs.

**Comment by cmanallen** [comment]
@aliu39 Can you update the pull description to describe why this is necessary, what its trying to accomplish, and the general flow of the implementation?  It will help with my review.

File: tests/sentry/toolbar/utils/test_csp.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    new_csp = csp_add_directive(csp, new_key, new_values)
+    dirs = get_directives(csp)
+    new_dirs = get_directives(new_csp)
+    assert new_key in new_dirs
+    for key in new_dirs:
+        if key != new_key:
+            assert key in dirs and new_dirs[key] == dirs[key]
+        else:
+            assert new_dirs[key] == dirs.get(key, set()).union(set(new_values))
+    return new_dirs
```
Comment (cmanallen) [inline tests/sentry/toolbar/utils/test_csp.py:23]: This is too complicated for me to understand.

Can we construct a CSP as a simple string, run the `csp_add_directive` function, and assert the result is equal to some string literal.

File: src/sentry/toolbar/views/base.py (hunk 1 above)
Comment (cmanallen) [reply src/sentry/toolbar/views/base.py:18]: Should this use the `sentry:toolbar_allowed_origins` option?

File: src/sentry/toolbar/views/iframe_view.py [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
 SUCCESS_TEMPLATE = "sentry/toolbar/iframe.html"
 INVALID_TEMPLATE = "sentry/toolbar/iframe-invalid.html"
 
+logger = logging.getLogger(__name__)
+
 
 @region_silo_view
-class IframeView(OrganizationView):
+class IframeView(ToolbarView):
     def respond(self, template: str, context: dict[str, Any] | None = None, status: int = 200):
```
Comment (cmanallen) [inline src/sentry/toolbar/views/iframe_view.py:21]: This class is overriding the respond method in the base class.  Should it be moved out of the base class and into the login view?

**Review by cmanallen — CHANGES_REQUESTED** [review]

File: src/sentry/toolbar/views/iframe_view.py (hunk 3 above)
Comment (aliu39) [reply src/sentry/toolbar/views/iframe_view.py:21]: We only need the `X-Frame-Options` for the iframe response.

File: src/sentry/toolbar/views/base.py (hunk 1 above)
Comment (ryan953) [reply src/sentry/toolbar/views/base.py:18]: I think it'll endup being the domain name(s) of whereever sentry is hosted. That option should be the list of 3rd party domains where their app is running

**Comment by aliu39** [comment]
> Should this use the sentry:toolbar_allowed_origins option?

Hm weird gh isn't letting me reply to this..
No, we don't want to expose the customer domains. It's also covered by `self` which I saw is included for prod responses.
In this case, `self` is the customer's browser/domain. The source of the inline script is Sentry.

**Comment by aliu39** [comment]
I'm going to rewrite this to use the [@csp_update decorator](https://django-csp.readthedocs.io/en/latest/decorators.html#csp-update). The default CSP directives are defined in django settings: 
https://github.com/getsentry/sentry/blob/4777d12cb2f2991277536dc3ddbc9cc42b50b829/src/sentry/conf/server.py#L457-L457
https://github.com/getsentry/getsentry/blob/4a9cd69d9a04ffd30bd8e3afebe45246eadfba19/getsentry/conf/settings/prod.py#L405-L405

File: src/sentry/toolbar/views/iframe_view.py (hunk 3 above)
Comment (cmanallen) [reply src/sentry/toolbar/views/iframe_view.py:21]: Oh I see I missed the call to super.  👍

**Comment by aliu39** [comment]
Reopening at https://github.com/getsentry/sentry/pull/78230 since it's much easier than reverting changes here. The decorator simplifies things a lot and decided we don't need the base class.

**Pull request closed without merging**